mm2src/coins/lightning.rs

```diff
@@ -389,6 +583,22 @@ pub async fn connect_to_lightning_node(ctx: MmArc, req: ConnectToNodeRequest) ->
     let (node_pubkey, node_addr) = parse_node_info(req.node_id.clone())?;
     let res = connect_to_node(node_pubkey, node_addr, ln_coin.peer_manager.clone()).await?;
 
+    // If a node that we have an open channel with changed it's address, "connect_to_lightning_node"
+    // can be used to reconnect to the new address while saving this new address for reconnections.
+    if let ConnectToNodeRes::ConnectedSuccessfully(_, _) = res {
+        let mut nodes_addresses = ln_coin.nodes_addresses.lock();
+        if let Entry::Occupied(mut entry) = nodes_addresses.entry(node_pubkey) {
```

Comment:
`nodes_addresses` are for the channels counterparties addresses only which we need to be always connected to. An address gets added to it in 2 cases. The first case is when opening a new channel and in this case, we get the address from the `open_channel` request. The second case is when restarting a lightning node and in this case, we get the addresses from the saved file.

`connect_to_lightning_node` can be used for 2 purposes. The main purpose is to connect to a node to get the network graph from it. In this case, no reconnection is needed since the network graph is saved and will be reloaded on restart. A second purpose was added for if a node we have an open channel with changes its address, this is the above case. In this case when a user finds that a channel is unusable from the `list_channels` response, using the counterparty_node_id the user can search on a lightning explorer for the node and get its new address. Using the new address the user can reconnect to the node using the `connect_to_lightning_node` RPC, if connected successfully, we should save the node's new address. That's why the `Entry::Occupied` case is only needed.

Reply:
Got it, thanks for the clarification!